app/models/compute_resources/foreman/model/openstack.rb

```diff
@@ -94,7 +107,7 @@ def boot_from_volume(args = {})
     end
 
     def create_vm(args = {})
-      boot_from_volume(args) if Foreman::Cast.to_bool(args[:boot_from_volume])
+      boot_from_volume(args) if (Foreman::Cast.to_bool(args[:boot_from_volume]) || Foreman::Cast.to_bool(args[:boot_from_snapshot]))
```

Comment:
Or reuse it and rename it to something more appropriate like `create_boot_volume` perhaps?